tests/scripts/all.sh

```diff
@@ -4340,13 +4340,18 @@ component_build_aes_aesce_armcc () {
 }
 
 support_build_sha_armce() {
-    # clang >= 4 is required to build with SHA extensions
-    clang_ver="$(clang --version|grep version|sed -E 's#.*version ([0-9]+).*#\1#')"
+    if ( $(which clang > /dev/null) ); then
```

Comment:
Why not do this without the sub-shell?

```
if which clang > /dev/null ; then
```

Reply:
Also, `$(which clang ...)` will try to run the `clang` that `which` finds

Reply:
Hmm, and what about using `command`?

```
if command -v clang > /dev/null ; then
```

Reply:
I didn't know about `command`, but unfortunately `bash command` is ungoogleable...

Reply:
Classic or POSIX way, which I'd also consider the idiomatic way in bash:
```
if type clang >/dev/null 2>/dev/null; then
```
Another POSIX way:
```
if command -v clang >/dev/null 2>/dev/null; then
```
[Don't use `which`](https://unix.stackexchange.com/questions/85249/why-not-use-which-what-to-use-then).